lib/app/modules/us_core_r4/clinicalnotes_sequence.rb

```diff
@@ -0,0 +1,228 @@
+# frozen_string_literal: true
+
+module Inferno
+  module Sequence
+    class UsCoreR4ClinicalNotesSequence < SequenceBase
+      group 'US Core R4 Profile Conformance'
+
+      title 'Clinical Notes Guideline Tests'
+
+      description 'Verify that DocumentReference and DiagnosticReport resources on the FHIR server follow the US Core R4 Clinical Notes Guideline'
+
+      test_id_prefix 'ClinicalNotes' # change me
+
+      requires :token, :patient_id
+      conformance_supports :DocumentReference, :DiagnosticReport
+
+      details %(
+
+        The #{title} Sequence tests DiagnosticReport and DocumentReference resources associated with the provided patient.  The resources
+        returned will be checked for consistency against the [US Core Clinical Notes Guidance](https://build.fhir.org/ig/HL7/US-Core-R4/clinical-notes-guidance.html)
+
+      )
+
+      @clinicalnotes_found = false
+
+      def test_clinicalnotes_documentreference(type_code)
+        skip 'No Clinical Notes appear to be available for this patient. Please use patients with more information.' unless @clinicalnotes_found
+
+        assert @must_have_type_code[type_code], "Clinical Notes shall have at least one DocumentReference with type #{type_code}"
+      end
+
+      def test_clinicalnotes_diagnosticreport(category_code)
+        skip 'No Clinical Notes appear to be available for this patient. Please use patients with more information.' unless @clinicalnotes_found
+
+        patient_val = @instance.patient_id
+        search_params = { 'patient': patient_val, 'category': category_code }
+
+        reply = get_resource_by_params(versioned_resource_class('DiagnosticReport'), search_params)
+        assert_response_ok(reply)
+        assert_bundle_response(reply)
+
+        resource_count = reply&.resource&.entry&.length || 0
+        @resources_found = true if resource_count.positive?
+
+        assert @resources_found, "Clinical Notes shall have at least one DiagnosticReport with category #{category_code}"
+
+        diagnosticreport_ary = reply&.resource&.entry&.map { |entry| entry&.resource }
+
+        diagnosticreport_ary&.each do |resource|
+          id = resource&.id
+
+          resource&.presentedForm&.each do |attachment|
+            url_value = attachment&.url
+            next if url_value.nil?
+
+            assert @attachment_url.key?(url_value), "Attachment #{url_value} referenced in DiagnosticReport/#{id} but not in any DocumentReference"
+            @attachment_url[url_value][:flag] = true
+          end
+        end
+      end
+
+      test 'Server returns expected results from DocumentReference search by patient+clinicalnotes' do
+        metadata do
+          id '01'
+          link 'https://build.fhir.org/ig/HL7/US-Core-R4/clinical-notes-guidance.html'
+          desc %(
+          )
+          versions :r4
+        end
+
+        patient_val = @instance.patient_id
+        category_val = 'clinical-note'
```

Comment:
`_val` doesn't really add any meaning to a variable name. How about `patient_id` and `category`?

Reply:
removed these two local variables